src/app/shared/services/latestSearchesUtil.js

```diff
@@ -0,0 +1,29 @@
+export default class LatestSearchesUtil {
+    constructor($window, latestSearches) {
+        this.$window = $window;
+        this.latestSearches = latestSearches;
+    }
+
+    getLatestSearches() {
+        return this.latestSearches = JSON.parse(this.$window.localStorage.getItem('latestSearches')) || [];
+    };
+
+    addCity(city) {
+        if (city == undefined || city == null || city.trim() === '' || this.latestSearches.includes(city)) {
```

Comment:
You can do this check `city == undefined || city == null || city.trim() === ''` with this `!city  || city.trim() === ''`

And be careful with this solution, Jennifer Null and her family doesn't like it. 
https://gizmodo.com/it-really-sucks-to-be-named-jennifer-null-1767159091